controllers/kafkatopic_controller.go

```diff
@@ -95,6 +106,19 @@ func (r *KafkaTopicReconciler) Reconcile(ctx context.Context, request reconcile.
 		return requeueWithError(reqLogger, "failed to lookup referenced cluster", err)
 	}
 
+	// Set managed status based on KafkaTopic managedBy annotation
+	managedByStatus := webhooks.TopicManagedByKoperatorAnnotationValue
+	if !isTopicManagedByKoperator(instance) {
+		managedByStatus = instance.GetAnnotations()[webhooks.TopicManagedByAnnotationKey]
+	}
+
+	if instance.Status.ManagedBy != managedByStatus {
```

Comment:
**Note**: for the CI errors you might need to release a new version of the API first and pull it into the root module, this is a recent change.

You can check with local replacement locally on your machine, also you can use a temporary dev tag or a real tag.

Reply:
Yes, thanks. First I want to get approve for this PR and later I will make the necessary changes on API and tagging.